sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/db/derived/DbDerivedDataRepository.kt

```diff
@@ -0,0 +1,72 @@
+package cash.z.ecc.android.sdk.internal.db.derived
+
+import cash.z.ecc.android.sdk.internal.model.EncodedTransaction
+import cash.z.ecc.android.sdk.internal.repository.DerivedDataRepository
+import cash.z.ecc.android.sdk.model.BlockHeight
+import cash.z.ecc.android.sdk.model.Transaction
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.toList
+
+@Suppress("TooManyFunctions")
+internal class DbDerivedDataRepository(
+    private val derivedDataDb: DerivedDataDb
+) : DerivedDataRepository {
+    override suspend fun lastScannedHeight(): BlockHeight {
+        return derivedDataDb.blockTable.lastScannedHeight()
+    }
+
+    override suspend fun firstScannedHeight(): BlockHeight {
+        return derivedDataDb.blockTable.firstScannedHeight()
+    }
+
+    override suspend fun isInitialized(): Boolean {
+        return derivedDataDb.blockTable.count() > 0
+    }
+
+    override suspend fun findEncodedTransactionById(txId: Long): EncodedTransaction? {
+        return derivedDataDb.transactionTable.findEncodedTransactionById(txId)
+    }
+
+    override suspend fun findNewTransactions(blockHeightRange: ClosedRange<BlockHeight>): List<Transaction> =
+        derivedDataDb.allTransactionView.getTransactionRange(blockHeightRange).toList()
+
+    override suspend fun findMinedHeight(rawTransactionId: ByteArray) = derivedDataDb.transactionTable
+        .findMinedHeight(rawTransactionId)
+
+    override suspend fun findMatchingTransactionId(rawTransactionId: ByteArray) = derivedDataDb.transactionTable
+        .findDatabaseId(rawTransactionId)
+
+    override suspend fun findBlockHash(height: BlockHeight) = derivedDataDb.blockTable.findBlockHash(height)
+
+    override suspend fun getTransactionCount() = derivedDataDb.transactionTable.count()
+
+    override fun invalidate() {
+        TODO("Not yet implemented")
```

Comment:
We still miss these 3 methods implementation - `invalidate`, `cleanupCancelledTx`, `deleteExpired`.